core/frontend/src/components/vehiclesetup/calibration.ts

```diff
@@ -0,0 +1,149 @@
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import {
+  MavCmd, MAVLinkType, MavResult,
+} from '@/libs/MAVLink2Rest/mavlink2rest-ts/messages/mavlink2rest-enum'
+import autopilot_data from '@/store/autopilot'
+import { sleep } from '@/utils/helper_functions'
+
+enum PreflightCalibration {
+  GYROSCOPE,
+  GYROSCOPE_TEMPERATURE,
+  MAGNETOMETER,
+  PRESSURE,
+  RC,
+  RC_TRIM,
+  ACCELEROMETER,
+  BOARD_LEVEL,
+  ACCELEROMETER_TEMPERATURE,
+  SIMPLE_ACCELEROMETER,
+  COMPASS_MOTOR_INTERFERENCE,
+  AIRPSEED,
+  ESC,
+  BAROMETER_TEMPERATURE,
+}
+
+class Calibrator {
+  private static instance: Calibrator
+
+  private calibrating: PreflightCalibration | undefined
+
+  private calibrationStatus: MavResult | undefined
+
+  constructor() {
+    mavlink2rest.startListening(MAVLinkType.COMMAND_ACK).setCallback((content) => {
```

Comment:
can we make it so this listener only lives while the calibration is being done?

start it at start() and discard() it when the calibration is successful/fails

Reply:
This is mostly a free websocket, it's handle by the server where it only sends the messages that match the regex. Since it's registered to COMMAND_ACK, it should have no performance impact.